sdb/commands/spl/internal/kmem_helpers.py

```diff
@@ -91,12 +92,23 @@ def object_size(cache: drgn.Object) -> int:
 def nr_objects(cache: drgn.Object) -> int:
     assert sdb.type_canonical_name(cache.type_) == 'struct spl_kmem_cache *'
     if backed_by_linux_cache(cache):
-        return int(cache.skc_obj_alloc.value_())
+        return obj_alloc(cache)
     return int(cache.skc_obj_total.value_())
 
 
 def obj_alloc(cache: drgn.Object) -> int:
     assert sdb.type_canonical_name(cache.type_) == 'struct spl_kmem_cache *'
+    try:
+        return int(drgn_percpu.percpu_counter_sum(cache.skc_linux_alloc))
+    except AttributeError:
```

Comment:
out of curiosity, is there no way to ask if the structure contains a given member? e.g. something like:
if 'skc_linux_alloc' in cache:
    return int(drgn_percpu.percpu_counter_sum(cache.skc_linux_alloc))
else:
    return int(cache.skc_obj_alloc.value_())
```

Reply:
Unfortunately I did look at drgn.Type API for that and there wasn't such a thing. The way to do this would be something like `if 'skc_linux_alloc` in [ x.name for x in type.members]`. I thought that the code for the loop was probably not worth it. Does this make sense? Do you think a variant of this alternative would be preferable?

Reply:
Thanks, got it. I'm good with what you have.